operators/src/processing/vector_join/mod.rs

```diff
@@ -0,0 +1,191 @@
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+use geoengine_datatypes::collections::VectorDataType;
+
+use crate::engine::{
+    ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedVectorOperator,
+    Operator, TypedVectorQueryProcessor, VectorOperator, VectorQueryProcessor,
+    VectorResultDescriptor,
+};
+use crate::error;
+use crate::util::Result;
+
+use self::equi_data_join::EquiLeftJoinProcessor;
+
+mod equi_data_join;
+
+/// The vector join operator requires two inputs and the join type.
+pub type VectorJoin = Operator<VectorJoinParams>;
+
+/// A set of parameters for the `VectorJoin`
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct VectorJoinParams {
+    #[serde(flatten)]
+    join_type: VectorJoinType,
+}
+
+/// Define the type of join
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(tag = "type")]
+pub enum VectorJoinType {
+    /// A left equi join between a `GeoFeatureCollection` and a `DataCollection`
+    EquiLeft {
+        left_column: String,
+        right_column: String,
+        /// which prefix to use if columns have conflicting names?
+        right_column_prefix: String,
```

Comment:
make this an `Option` and have some default behavior like (1) and (2)? Would make usage of operator easier for user

Reply:
b44c58d

Reply:
using a hard coded english word makes it hard for i18n later on. That's why I would prefer something more universal like `(2)`, `(*)` suffix or something like that. Also `right` is a rather technical term which might not be obvious for non-database users. If it has to be an english prefix I would rather take something like `other` or `(conflict)`